### PR TITLE
POC: use `Sidekiq.perform_bulk` for Github repos

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Optimization

## Description

**tl;dr:**
This PR changes our `GithuRepo` model to use Sidekiq's recently added `push_bulk` method. If this works as expected we can then change similar code to use the same method.

**Full context:**
For a while now, Sidekiq had a [`push_bulk` method](https://www.rubydoc.info/github/mperham/sidekiq/Sidekiq%2FClient:push_bulk) which allows pushing large numbers of jobs to Sidekiq. The goal is to reduce Redis roundtrip latency by replacing N calls with 1. The recommended number of jobs to push was < 1000.

Sidekiq 6.3.0 added a higher-level wrapper around this, [`perform_bulk`](https://www.rubydoc.info/gems/sidekiq/Sidekiq%2FWorker%2FSetter:perform_bulk). This encodes the best practice of 1000 jobs per call though one can change that number via the `batch_size` argument.

**Further reading:**

* [Sidekiq Wiki: Bulk Queuing](https://github.com/mperham/sidekiq/wiki/Bulk-Queueing)
* [One Ruby Thing: Enqueue Jobs Quickly with Sidekiq’s Bulk Features](https://andycroll.com/ruby/enqueue-jobs-quickly-with-sidekiq-bulk/)

## QA Instructions, Screenshots, Recordings

Specs should pass.

### UI accessibility concerns?

No UI changes

## Added/updated tests?

- [X] Yes

## [Forem core team only] How will this change be communicated?

- [X] I will share this change internally with the appropriate teams

### CODE REVIEW

File: app/models/github_repo.rb [hunk 1]
```diff
@@ -27,9 +27,8 @@ def self.upsert(user, **params)
   end
 
   def self.update_to_latest
-    where("updated_at < ?", 26.hours.ago).ids.each do |repo_id|
-      GithubRepos::RepoSyncWorker.perform_async(repo_id)
-    end
+    ids = where("updated_at < ?", 26.hours.ago).ids.each_slice(1).to_a
```
Comment (citizen428) [inline app/models/github_repo.rb:30]: Reading note: `perform_bulk` expects its arguments as an array of arrays, where each inner array contains the arguments for the job it calls. In this case here, we need to provide the repo ids like this: `[[1], [2], ... ]`.

Comment (jgaskins) [inline app/models/github_repo.rb:30]: This is slick, but feels a bit too clever. Can we clarify it some?

```suggestion
    ids = where("updated_at < ?", 26.hours.ago).ids.map do |id|
      [id]
    end
```

It takes more characters, but I think it's clearer that it's wrapping each `id` in an inner array.

**Review by jgaskins — APPROVED** [review]
This seems like a nice, low-risk place to test this since it only runs in a scheduled job.

**Comment by jgaskins** [comment]
I'll keep an eye on [this worker's stats](https://app.datadoghq.com/apm/resource/sidekiq/sidekiq.job/4fb12e4910b5b833?query=env%3Aproduction%20service%3Asidekiq%20operation_name%3Asidekiq.job%20resource_name%3A%22GithubRepos%3A%3AUpdateLatestWorker%22&env=production&index=apm-search&topGraphs=latency%3Alatency%2Cerrors%3Aversion_count%2Chits%3Acount%2CbreakdownAs%3Apercentage&start=1643038600909&end=1643125000909&paused=false).

File: app/models/github_repo.rb (hunk 1 above)
Comment (citizen428) [reply app/models/github_repo.rb]: Fair. I'm pretty sure I copy/pasted this off the Sidekiq wiki, agreed it's not the easiest to read. I didn't accept your suggestion because I realized we can also simplify the `where` to use a beginless range (`where(updated_at: ...26.hours.ago)`).

**Comment by citizen428** [comment]
@jgaskins I updated the code now. Since it's 11:24pm here I didn't feel like waiting for the Travis run to finish, so I added the `merge by any core` label in case you feel like merging it.
